proto/spacemesh/global_state_types.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+package spacemesh;
+import "spacemesh/types.proto";
+
+// Data items related to an account
+enum AccountDataFlag {
+    NONE = 0;
+    TRANSACTION_RECIEPT = 1; // tx receipt for a tx to or from an account
+    REWARD = 2; // reward awarded to an account
+}
+
+message AccountDataParams {
+    AccountId account_id = 1;
+    uint32 account_data_bit_field = 2; // A bit field of AccountDataFlag
+}
+
+message AccountDataQueryParams {
+    AccountDataParams account_data = 1;
+    uint32 max_results = 2; // max numbers of results client would like to get
+    uint32 offset = 3; // return results from offset
+}
+
+// An account related data item
+message AccountData {
+    oneof item {
+        Reward reward = 1;
+        TransactionReceipt receipt = 2;
+        Account account = 3;
+    }
+}
+
+message AccountDataQueryResults {
+    repeated AccountData account_item = 1;
+    uint32 total_results = 2;
+}
+
+message SmesherDataQueryParams {
+    SmesherId smesher_id = 1;
+    uint32 max_results = 2; // max numbers of results client would like to get
+    uint32 offset = 3;      // return results from offset
+}
+
+message SmesherDataQueryResults {
+    repeated Reward rewards = 1;
+    uint32 total_results = 2;
+}
+
+enum GlobalStateEntityFlag {
+    NONE = 0;
+    REWARD = 1;
+    TRANSACTION_STATE = 2;
+    TRANSACTION_RECIEPT = 4;
+    GLOBAL_STATE = 8;
+}
+
+message GlobalStateEntities {
+    uint32 flags = 1; // bitmap of GlobalStateEntityFlags
+}
+
+message GlobalStateEntity {
```

Comment:
I don't think `Entity` is the right word here - maybe `Entry`? Or `Object` or `Datum`

Reply:
I ran out of good names today. These are global state entities. I think entity describes these better than the other names you propose. Maybe `Item` or `DataItem` is better?